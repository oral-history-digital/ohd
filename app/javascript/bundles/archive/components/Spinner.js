import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import spinnerSrc from 'images/large_spinner.gif';

export default function Spinner({
    withPadding = false,
    className,
    style,
}) {
    return (
        <img
            src={spinnerSrc}
            className={classNames('Spinner', className, {
                'Spinner--withPadding': withPadding,
            })}
            style={style}
            alt=""
        />
    );
}

Spinner.propTypes = {
    withPadding: PropTypes.bool,
    className: PropTypes.string,
    style: PropTypes.object,
};
