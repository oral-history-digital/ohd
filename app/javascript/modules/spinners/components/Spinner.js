import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './Spinner.module.scss';
import spinnerSrc from './large_spinner.gif';

export default function Spinner({
    small = false,
    withPadding = false,
    className,
    style,
}) {
    return (
        <img
            src={spinnerSrc}
            className={classNames(styles.img, className, {
                [styles.withPadding]: withPadding,
                [styles.small]: small,
            })}
            style={style}
            alt=""
        />
    );
}

Spinner.propTypes = {
    small: PropTypes.bool,
    withPadding: PropTypes.bool,
    className: PropTypes.string,
    style: PropTypes.object,
};
