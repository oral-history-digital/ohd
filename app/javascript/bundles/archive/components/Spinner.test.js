import React from 'react';
import renderer from 'react-test-renderer';

import Spinner from './Spinner';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <Spinner
                className="spinner"
                style={{ margin: '1rem' }}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders padding CSS class', () => {
    const tree = renderer
        .create(
            <Spinner withPadding />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
