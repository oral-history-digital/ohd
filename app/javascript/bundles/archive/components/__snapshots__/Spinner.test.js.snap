// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt=""
  className="Spinner spinner"
  src="mock.jpg"
  style={
    Object {
      "margin": "1rem",
    }
  }
/>
`;

exports[`renders padding CSS class 1`] = `
<img
  alt=""
  className="Spinner Spinner--withPadding"
  src="mock.jpg"
/>
`;
